paddle/fluid/memory/detail/meta_cache.cc

```diff
@@ -22,16 +22,17 @@ namespace detail {
 
 MetadataCache::MetadataCache(bool uses_gpu) : uses_gpu_(uses_gpu) {}
 
-MemoryBlock::Desc MetadataCache::load(const MemoryBlock* block) const {
+MemoryBlock::Desc* MetadataCache::load_desc(const MemoryBlock* block) const {
   if (uses_gpu_) {
     auto existing_desc = cache_.find(block);
     PADDLE_ENFORCE_EQ(existing_desc->second.check_guards(), true);
-    return existing_desc->second;
+    return const_cast<MemoryBlock::Desc*>(&(existing_desc->second));
```

Comment:
Do not use const_cast. Use overloaded functions instead.

Reply:
Done